capplets/about-me/e-image-chooser.c

```diff
@@ -137,6 +137,8 @@ e_image_chooser_init (EImageChooser *chooser)
 	priv = chooser->priv = g_new0 (EImageChooserPrivate, 1);
 
 	priv->image = gtk_image_new ();
+	priv->image_width = 100;
+	priv->image_height = 100;
```

Comment:
No, it won't... You don't make such changes without explaining why it is needed! All I see:
 `Always keep image size to 100x100`

Why? I see other code that assigns `image_width`... Does that means that `Always keep` part is wrong or there is obsolete code that should be removed? Do I get these answers when I run `mate-about-me` application? I doubt...

Look at this: https://github.com/mate-desktop/mate-control-center/blob/6d7174f1e2bd15f741f58cf144d9d509fb215447/capplets/about-me/e-image-chooser.c#L239-L251

Why code still calculates `new_width` and `new_height` if it does not use it? Compiler might even warn about it if needed warnings are turned on...

Why you subscribe 5 from 100 if you want keep image 100x100? Why don't you use 95x95 then to avoid subtraction from constant size?

And to speak truth I would even say that it is unrelated change to account service and should not be in this pull request!

Can you spot these mistakes from run-time testing? If you ignore such things code quality will decrease over time and sooner or later it will turn into unmaintainable piece of shit. And that is your, as maintainer, job to make sure that it does not happen, no?

Of course I just commented and if you disagree and/or if it is good for you you can merge it! But keeps this in mind if you will ever want to use `metacity` as main wm and will want to contribute to it. That is how I review and how I treat maintainers job.

Reply:
Exactly - didn't notice... Look at `set_image_from_data` function after his changes. 

https://github.com/mate-desktop/mate-control-center/blob/6d7174f1e2bd15f741f58cf144d9d509fb215447/capplets/about-me/e-image-chooser.c#L240-L243

This can be deleted without thinking, no?

Then why would you calculate `scale` if you don't really use. You don't need to calculate `scale` to decide if scaling is needed or image is at right size already.

https://github.com/mate-desktop/mate-control-center/blob/6d7174f1e2bd15f741f58cf144d9d509fb215447/capplets/about-me/e-image-chooser.c#L213-L216

Is not this "dead" code now? Or can you tell me how `image_width` could become `0`?

Reply:
Why is 100? Because in the [mate-about-me.c](https://github.com/mate-desktop/mate-control-center/blob/master/capplets/about-me/mate-about-me.c#L41) file, the size of the button defined as MAX_HEIGHT and MAX_WIDTH are 100.

Why minus 5? Because the [original code](https://github.com/mate-desktop/mate-control-center/blob/master/capplets/about-me/e-image-chooser.c#L205), minus 5, to keep the picture and border between 5 pixels position.

Of course, there are problems in the current code at master branch, which can be found through testing.

For example, if you set different images one after another as user face, even if these images have the same size, they will still cause the displayed images to grow larger and remain 5 pixels at a time. If you use larger sized pictures, the whole dialog box will become larger.

e-image-chooser.c implements a general image selection widget, it is ok, but it is not suitable for mate-about-me, at least in this interface, and its code needs to be modified to keep the user's face image the same visual size, just add e-image-chooser.c(h) is not enough.

When I let mate-about-me to support accounts service, I found this problem, and I wanted to make it work as I expected with minimal modifications. This creates dead code, but this is not my problem. I at least let it work as expected.

Reply:
If had to strictly adhere to one commit to solve only one problem, I could remove the modifications here in this RP, and create another RP to solve the bug, and clean up the dead code in e-image-chooser.c.

Reply:
I guess before you do any other extra work, wait for response from MATE developers (I am not).

If it would be up to me then I would **require** that there are separate pull requests for unrelated changes. Adding accounts service support and changing image scaling code / fixing image size bug are completely different things.

>This creates dead code, but this is not my problem.

Tell me then - whose problem it is then?

Reply:
> Tell me then - whose problem it is then?

Yes, I did not express my understanding correctly. It should not be modified here, and it will be modified later.